src/store/generator/slices/thresholds.ts

```diff
@@ -0,0 +1,36 @@
+import { Threshold } from '@/types/thresholds'
+import { ImmerStateCreator } from '@/utils/typescript'
+
+interface State {
+  thresholds: Array<Threshold>
+}
+
+interface Actions {
+  addThreshold: (threshold: Threshold) => void
+  updateThreshold: (threshold: Threshold) => void
+  deleteThreshold: (id: string) => void
+}
```

Comment:
Initial state and actions to get some UI and validation working in the upcoming branch.